orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -1081,9 +1087,26 @@ def _load_model_states(self, state_dict, strict):
     def _load_optimizer_states(self, current_state_dict, state_dict):
         """Load the optimizer states onto the training session state dictionary"""
 
+        def _check_optimizer_mismatch(state_dict):
+            """Assert that the loaded optimizer has the same config as the current training session config"""
+
+            # the state_dict optimizer_name can be a byte string (if coming from checkpoint file)
+            # or can be a regular string (coming from user)
+            optimizer_name = \
+                state_dict[_utils.state_dict_trainer_options_key()][_utils.state_dict_trainer_options_optimizer_name_key()]
+            try:
+                optimizer_name = optimizer_name.decode()
+            except AttributeError:
+                pass
```

Comment:
Why is this ok? document the reason

Reply:
same reason as earlier. 
```optimizer_name``` can be either a byte string or a regular string. I want it to be a regular string. So, I try to decode it from a byte string. If not successfult due to ```AttributeError```, then it already is a regular string, so no need to convert.